core/web/resolver/errors.go

```diff
@@ -16,19 +16,30 @@ const (
 )
 
 type NotFoundErrorUnionType struct {
-	err     error
-	message string
+	err               error
+	message           string
+	isExpectedErrorFn func(err error) bool
 }
 
 // ToNotFoundError resolves to the not found error resolver
 func (e *NotFoundErrorUnionType) ToNotFoundError() (*NotFoundErrorResolver, bool) {
-	if e.err != nil && errors.Is(e.err, sql.ErrNoRows) {
+	isErrFn := isNotFoundSQLError
+
+	if e.isExpectedErrorFn != nil {
+		isErrFn = e.isExpectedErrorFn
+	}
+
+	if e.err != nil && isErrFn(e.err) {
```

Comment:
Should we enforce `isErrFn` returning `false` if `err == nil`? There isn't really a good way to enforce such a postcondition in Go though, so maybe we should keep the `e.err != nil` check.

Reply:
I think we could maybe do that, but then there's no guarantee that someone could implement `isExpectedErrorFn` without actually checking that `e.err` is not nil... the way that I found decent for this was to just allow the this part of the code to check for the existence of the error.